pkg/ring/token_file.go

```diff
@@ -0,0 +1,79 @@
+package ring
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+	"sort"
+)
+
+type TokenFile struct {
+	PreviousState InstanceState `json:"previousState"`
```

Comment:
Do we need to make it omit empty or it is fine to leave it here?
Is a cortex with old image able to read the token file using the new format?

Reply:
I also added compatibility unit tests.